e2e/testcases/namespace_strategy_test.go

```diff
@@ -0,0 +1,211 @@
+// - Set namespaceStrategy: implicit - should create the namespace implicitly
+// - Set namespaceStrategy: explicit - namespace still exists but is unmanaged
+func TestNamespaceStrategy(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.OverrideAPI, ntopts.Unstructured)
+
+	rootSyncNN := nomostest.RootSyncNN(configsync.RootSyncName)
+	rootReconcilerNN := core.RootReconcilerObjectKey(rootSyncNN.Name)
+	rootSync := fake.RootSyncObjectV1Alpha1(rootSyncNN.Name)
+	// set the NamespaceStrategy to explicit
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"namespaceStrategy": "explicit"}}}`)
+	err := nt.Watcher.WatchObject(
+		kinds.Deployment(), rootReconcilerNN.Name, rootReconcilerNN.Namespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(
+				reconcilermanager.Reconciler,
+				reconcilermanager.NamespaceStrategy,
+				string(configsync.NamespaceStrategyExplicit),
+			),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	// add a resource for which the namespace is not declared/created
+	fooNamespace := fake.NamespaceObject("foo-implicit")
+	cm1 := fake.ConfigMapObject(core.Name("cm1"), core.Namespace(fooNamespace.Name))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/cm1.yaml", cm1))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add cm1"))
+	// check for error
+	nt.WaitForRootSyncSyncError(rootSyncNN.Name, applier.ApplierErrorCode,
+		"failed to apply ConfigMap, foo-implicit/cm1: namespaces \"foo-implicit\" not found")
+	// switch the mode to implicit
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"namespaceStrategy": "implicit"}}}`)
+	// check for success
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	// assert that implicit namespace was created
+	err = nt.Validate(fooNamespace.Name, fooNamespace.Namespace, &corev1.Namespace{},
+		testpredicates.HasAnnotation(common.LifecycleDeleteAnnotation, common.PreventDeletion),
+		testpredicates.HasAnnotation(metadata.ResourceManagementKey, metadata.ResourceManagementEnabled),
+		testpredicates.HasAnnotation(metadata.ResourceManagerKey, string(declared.RootReconciler)),
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	// switch mode back to explicit
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"namespaceStrategy": "explicit"}}}`)
+	// assert that namespace is no longer managed
+	err = nt.Watcher.WatchObject(kinds.Namespace(), fooNamespace.Name, fooNamespace.Namespace,
+		[]testpredicates.Predicate{
+			// still has PreventDeletion
+			testpredicates.HasAnnotation(common.LifecycleDeleteAnnotation, common.PreventDeletion),
+			// management annotations should be removed
+			testpredicates.MissingAnnotation(metadata.ResourceManagementKey),
+			testpredicates.MissingAnnotation(metadata.ResourceManagerKey),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+// This test case is focused on the behavior of using namespaceStrategy: explicit
```

Comment:
Comment needs to start with TestNamespaceStrategyMultipleRootSyncs

Reply:
surprised it passed linting without this...

Reply:
I don't think the linter requires for test functions to have the test name in the docstring